src/main/scala/com/gu/salesfoce/messageHandler/MessageHandlerLambda.scala

```diff
@@ -0,0 +1,111 @@
+      |	<soapenv:Body>
+      |		<notificationsResponse xmlns="http://soap.sforce.com/2005/09/outbound">
+      |			<Ack>true</Ack>
+      |		</notificationsResponse>
+      |	</soapenv:Body>
+      |</soapenv:Envelope>
+    """.stripMargin
+
+  val okResponse = ApiResponse("200", Headers(), okXml)
+
+  def parseMessage(requestBody: String): List[ContactNotification] = {
+    val is = new ByteArrayInputStream(requestBody.getBytes)
+    val messageFactory = MessageFactory.newInstance()
+    val soapMessage = messageFactory.createMessage(null, is)
+    val body = soapMessage.getSOAPBody
+    val jc = JAXBContext.newInstance(classOf[Notifications])
+    val unmarshaller = jc.createUnmarshaller()
+    val je = unmarshaller.unmarshal(body.extractContentAsDocument(), classOf[Notifications])
+    val notifications = je.getValue()
+    notifications.getNotification.toList
+  }
+
+  case class QueueMessage(contactId: String)
+
+  implicit val messageFormat = Json.format[QueueMessage]
+
+  def credentialsAreValid(inputEvent: JsValue): Boolean = {
+
+    val maybeApiClientId = (inputEvent \ "queryStringParameters" \ "apiClientId").asOpt[String]
+    val maybeApiClientToken = (inputEvent \ "queryStringParameters" \ "apiToken").asOpt[String]
+    val maybeCredentials = (maybeApiClientId, maybeApiClientToken)
+    maybeCredentials match {
+      case (Some(apiClientId), Some(apiToken)) => {
+        (apiClientId == Config.apiClientId && apiToken == Config.apiToken)
+      }
+      case _ => {
+        logger.info(s"Could not find credentials in request")
+        false
+      }
+    }
+  }
+
+  def sendToQueue(notification: ContactNotification): Future[Try[SendMessageResult]] = {
+    val queueMessage = QueueMessage(notification.getSObject.getId)
+    val queueMessageString = Json.prettyPrint(Json.toJson(queueMessage))
+    queueClient.send(queueName, queueMessageString)
+  }
+
+  def handleRequest(inputStream: InputStream, outputStream: OutputStream, context: Context): Unit = {
+
+    logger.info(s"Salesforce message handler lambda ${Config.stage} is starting up...")
+    val inputEvent = Json.parse(inputStream)
+    if (!credentialsAreValid(inputEvent)) {
+      logger.info("Request from Zuora could not be authenticated")
+      outputForAPIGateway(outputStream, unauthorized)
+    } else {
+      logger.info("Authenticated request successfully...")
+      val body = (inputEvent \ "body").as[String]
+      val notifications = parseMessage(body)
+      logger.info(s"found ${notifications.size} contact(s) in the salesforce xml")
+      val FutureResponses = notifications.map(sendToQueue)
+      val future = Future.sequence(FutureResponses).map { responses =>
+        val errors = responses collect { case Failure(error) => error }
+        if (errors.nonEmpty) {
+          errors.foreach(error =>
+            logger.error(s"error while trying to send message to queue", error))
+          outputForAPIGateway(outputStream, internalServerError)
```

Comment:
So if Salesforce sends us 3 contacts, and we fail to put 1/3 onto the SQS queue, we respond with a 500? I believe Salesforce will then re-try the whole message, which might mean we put the same contacts on the queue twice in quick succession - presumably we can handle that case in the code which is consuming from this queue?